### ISSUE TITLE
`attribute.QuoteType` does not output the correct quote type

### ISSUE THREAD

**Issue by rvishruth** [zzzprojects/html-agility-pack#560]
## 1. Description
`attribute.QuoteType` does not output the correct quote type 

In the code below, despite setting the GlobalAttributeValueQuote property to AttributeValueQuote.Initial, the `a.QuoteType` outputs the quote types as DoubleQuote, SingleQuote, and DoubleQuote, instead of the expected None, SingleQuote, and DoubleQuote.

**Follow up question:** Is there any way currently to detect the quote type of an attribute's value (even if it's unquoted to begin with)?

## 2. Exception
N/A

## 3. Fiddle or Project
```
// @nuget: HtmlAgilityPack
using System;
using HtmlAgilityPack;
using System.Linq;

public class Program
{
	public static void Main()
	{
		// Load
		string htmlString = "<p><a href=https://example.com>Link</a><a href='https://example.com'>Link</a><a href=\"https://example.com\">Link</a></p>";
		HtmlDocument dom = new HtmlDocument()
		{
			GlobalAttributeValueQuote = AttributeValueQuote.Initial
		};
		dom.LoadHtml(htmlString);
		var documentNodeDescendants = dom.DocumentNode.Descendants().ToList();
		Console.WriteLine(String.Join(",", documentNodeDescendants.SelectMany(n => n.Attributes).Select(a => a.QuoteType)));
	}
}
```

Output:
```
DoubleQuote,SingleQuote,DoubleQuote
```

![image](https://github.com/zzzprojects/html-agility-pack/assets/79176273/0a79a1cb-55e2-4d67-8a85-4c8638b83e36)

## 4. Any further technical details
- HAP version: 1.11.59
- NET version: .NET8

**Comment by JonathanMagnan** [comment]
Hello @rvishruth ,

Thank you for reporting.

On the good side, we output correctly without a quote when performing `Console.WriteLine(dom.DocumentNode.InnerHtml);`

We will look at this issue.

> Follow up question: Is there any way currently to detect the quote type of an attribute's value (even if it's unquoted to begin with)?

I'm not sure I understand your follow-up question. Could you try again?

Best Regards,

Jon

**Comment by rvishruth** [comment]
Thanks Jon! Sorry let me rephrase -

What is the current best way to detect if an attribute's value is unquoted?

**Comment by JonathanMagnan** [comment]
Thank @rvishruth , now it's 100% clear.

You can currently get it through reflection:

```csharp
var internalQuoteTypeProperty = typeof(HtmlAttribute).GetProperty("InternalQuoteType", System.Reflection.BindingFlags.Public | System.Reflection.BindingFlags.NonPublic | System.Reflection.BindingFlags.Instance);

Console.WriteLine(String.Join(",", documentNodeDescendants.SelectMany(n => n.Attributes).Select(a => internalQuoteTypeProperty.GetValue(a))));
```

The `InternalQuoteType` should always be the one we parsed. We probably added it in the past for not impacting projects that were already using the QuoteType to be able to support the `Initial` flag.

Let me know if that solution could work for you. If yes, make sure to keep the `BindingFlags.Public` as we never know if we will make it public in the future.

Best Regards,

Jon

**Comment by rvishruth** [comment]
Thank you @JonathanMagnan! This makes sense!
